client/my-sites/checkout/checkout/checkout.jsx

```diff
@@ -130,7 +130,6 @@ export class Checkout extends React.Component {
 
 		if ( ! isEqual( previousCart, nextCart ) ) {
 			this.redirectIfEmptyCart();
-			this.setState( { previousCart: nextCart } );
```

Comment:
AFAICS, this is the only occurrence that sets `previousCart` state in this component. Removing it means that all references to `this.state.previousCart` become equivalent to its `previousCart`'s `initialState` -- `null`, which further means that we could entirely replace all references to `this.state.previousCart` with `null`. This is likely going to be semantically different from what we have now, and _might_ change the component's behavior. Probably best to think through what such a change would mean.

Storing a value called `previousCart` in `state` seems a bit like a code smell anyway, so there's a chance that either a) removing it won't really change anything, or b) it can be replaced with something more semantic.